test/version_solver_test.dart

```diff
@@ -1134,9 +1134,89 @@ void dartSdkConstraint() {
 
     await expectResolves(
         environment: {'_PUB_TEST_SDK_VERSION': '2.0.0'},
-        error: 'Package foo requires SDK version <2.0.0-dev.infinity but the '
+        error: 'Package foo requires SDK version <2.0.0 but the '
             'current SDK is 2.0.0.');
   });
+
+  test("prerelease override requires major sdk versions to match", () async {
```

Comment:
"prerelease" -> "pre-release" (here and below).

"sdk" -> "SDK", of course. :)

Reply:
done